src/TraceEvent/EventSources/ProcessMetadataEventSource.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Tracing
+{
+    [EventSource(Name = "ProcessMetadataEventSource")]
+    public sealed class ProcessMetadataEventSource : EventSource
+    {
+        public class Tasks
+        {
+            public const EventTask Process = (EventTask)1;
+            public const EventTask Thread = (EventTask)2;
+            public const EventTask Module = (EventTask)3;
+        }
+
+        [Event(1, Opcode = EventOpcode.Start, Task = Tasks.Process)]
+        public void ProcessStart(long ProcessId, long ParentProcessId, string Executable, string CommandLine)
+        {
+            this.WriteEvent(1, ProcessId, ParentProcessId, Executable, CommandLine);
+        }
+
+        [Event(2, Opcode = EventOpcode.Stop, Task = Tasks.Process)]
+        public void ProcessExit(long ProcessId, long ParentProcessId, int ExitCode, string Executable, string CommandLine)
+        {
+            this.WriteEvent(2, ProcessId, ParentProcessId, ExitCode, Executable, CommandLine);
+        }
+
+        [Event(3, Opcode = EventOpcode.Start, Task = Tasks.Thread)]
+        public void ThreadCreate(long ProcessId, long ThreadId, ulong StackBaseAddress, string ThreadName)
```

Comment:
Rundown events or capture state events would have the same format, so I thought we could overload this, but I can also just add a specific thing for "Rundown", or maybe a more neutral term for catch up events.

Reply:
I do think it would be worth knowing if the thread has existed longer than the trace or not.  Perhaps something similar to Thread/DCStart?  Also, if you don't mind, let's change to Thread/Start and Thread/Stop top match existing precedent.